interface/src/app/gene/genes.js

```diff
@@ -0,0 +1,384 @@
+        var previousQueries = queries.length;
+        for (var i = 0; i < data.length; i++) {
+          query = data[i].search;
+          if (!searchResults[query]) {
+            //if search term didn't already exist
+            searchResults[query] = data[i]; //add it to the results
+            queries.push(query); // add to the list of queries
+          }
+        }
+        if (previousQueries !== queries.length) {
+          $rootScope.$broadcast('results.update');
+        }
+        $rootScope.$broadcast('results.searchResultsReturned');
+      });
+    }
+  };
+}])
+
+.factory('SelectedGenesFactory', function() {
+  var selected_genes = {};
+
+  return {
+    clear: function() {
+      selected_genes = {};
+    },
+
+    addGene: function(geneObj) {
+      selected_genes[geneObj.id] = geneObj;
+    },
+
+    removeGene: function(geneObj) {
+      delete selected_genes[geneObj.id];
+    },
+
+    return_genes: function() {
+      return selected_genes;
+    }
+  };
+})
+
+// Directive for whole gene search form
+.directive('geneSearchForm', ['SearchResults', function(SearchResults) {
+  return {
+    controller: ['$scope', '$rootScope', 'SearchResults',
+      function($scope, $rootScope, SearchResults) {
+
+        $scope.loadingSearchResults = false;
+
+        // Clear any existing search results
+        SearchResults.clear();
+
+        $scope.searchGenes = function() {
+          if (!$scope.genesToAdd) { //if the query is empty
+            return false;
+          }
+          $scope.loadingSearchResults = true;
+
+          $rootScope.$broadcast('results.loadingSearchResults');
+          qparams = {
+            'query': $scope.genesToAdd.query
+          };
+
+          if ($scope.organism) {
+            qparams['organism'] = $scope.organism;
+          }
+
+          SearchResults.search(qparams);
+        };
+
+        $scope.$on('results.searchResultsReturned', function() {
+          $scope.loadingSearchResults = false;
+        });
+
+      }
+    ],
+    replace: true,
+    restrict: "E",
+    scope: {
+      query: '@'
+    },
+    templateUrl: 'gene/gene-search-form.tpl.html'
+  };
+}])
+
+// Directive for table containing search results
+.directive('searchResultTable', ['SearchResults', function(SearchResults) {
+  return {
+    controller: ['$scope', 'SearchResults', function($scope, SearchResults) {
+      $scope.currentPage = 1;
+      $scope.itemsPerPage = 10;
+      $scope.totalResults = 0;
+      $scope.maxSize = 10;
+      $scope.resultsForPage = [];
+      $scope.searchResults = SearchResults.getSearchResults();
+      $scope.loadingSearchResults = false;
+
+      $scope.addAllNonAmbiguous = function() {
+        // Function to automatically add all genes that
+        // only have one search result.
+        var searchResults = SearchResults.getSearchResults();
+        for (var key in searchResults) {
+          var results = searchResults[key];
+          if (results['found'].length <= 1) {
```

Comment:
JavaScript does not have **<==** or **>==** operators (surprise!). I think the reason behind it is that *<==** is equivalent to **NOT >**, and *>==** to **NOT <**. See more details here: